koku/api/report/queries.py

```diff
@@ -1267,6 +1267,10 @@ def order_by(self, data, order_fields):
             if field in numeric_ordering:
                 sorted_data = sorted(sorted_data, key=lambda entry: (entry[field] is None, entry[field]),
                                      reverse=reverse)
+            elif 'tag:' in field:
+                tag = field[4:]
+                sorted_data = sorted(sorted_data, key=lambda entry: entry[tag].lower(),
+                                     reverse=reverse)
```

Comment:
Here is the actual order-by tag code. All we do is take `tag:my-cool-tag` and strip off the `tag:` bit, then order alphabetically.

In the future, we can add more logic for distinguishing between numerals and strings, if required.

Reply:
Right now the UI restricts to grouping by a single tag, so this example is slightly contrived, but I hit 

```
/local/v1/reports/aws/costs/?filter[time_scope_value]=-1&group_by[tag:app]=*&group_by[tag:environment]=*&order_by[tag:environment]=asc 
```

and got a traceback, used pdb to see where we were hitting the None. With the double group by here, we get cases where there is a match for 1 tag, but the other tag comes back `None`. 

```
[2019-05-06 13:54:09,184] INFO: API: /local/v1/reports/aws/costs/?filter[time_scope_value]=-1&group_by[tag:app]=*&group_by[tag:environment]=*&order_by[tag:environment]=asc -- ACCOUNT: 10001 USER: user_dev
> /Users/aberglund/redhat/cmaas/koku/koku/api/report/queries.py(1272)order_by()
-> tag = field[4:]
(Pdb) n
(Pdb) print(tag)
environment
(Pdb) sorted_data[0]
{'app': 'analytics', 'environment': None, 'date': '2019-05', 'cost_units': 'USD', 'infrastructure_cost': Decimal('121.304848155'), 'derived_cost': Decimal('0'), 'cost': Decimal('121.304848155')}
(Pdb) n
> /Users/aberglund/redhat/cmaas/koku/koku/api/report/queries.py(1274)order_by()
-> reverse=reverse)
(Pdb) n
AttributeError: 'NoneType' object has no attribute 'lower'
```